sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrumentBuilder.java

```diff
@@ -51,7 +58,32 @@ public BuilderT setDescription(String description) {
   }
 
   private InstrumentDescriptor makeDescriptor(InstrumentType type, InstrumentValueType valueType) {
-    return InstrumentDescriptor.create(instrumentName, description, unit, type, valueType);
+    String validUnit = this.unit;
+    if (!isValidUnit(this.unit)) {
+      validUnit = DEFAULT_UNIT;
+      logger.log(
+          Level.WARNING,
+          "Instrument "
+              + instrumentName
+              + " unit \""
+              + this.unit
+              + "\" is invalid, using "
+              + DEFAULT_UNIT
+              + " instead. Instrument unit must be 63 or less ASCII characters.");
+    }
+    return InstrumentDescriptor.create(instrumentName, description, validUnit, type, valueType);
+  }
+
+  /**
+   * Determine if the instrument unit is valid. Units must be non-null, non-blank, strings of 63 or
+   * less ASCII characters.
+   */
+  // Visible for testing
+  static boolean isValidUnit(String unit) {
+    return unit != null
+        && !unit.equals("")
+        && unit.length() < 64
+        && StandardCharsets.US_ASCII.newEncoder().canEncode(unit);
```

Comment:
This probably needs clarification in the spec but I wouldn't allow control characters be allowed if restricting. Alternatively, given description is descriptive, I'm wondering why it's not a UTF-8 string.

Reply:
Perhaps I misunderstand the comment. UTF-8 characters are allowed for description, but unit is more restrictive. Agree with point about the control characters. Opened this [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2496) in the spec to address.

Reply:
Ooops, got unit and description mixed up